nemo/agents/voice_agent/utils/config_manager.py

```diff
@@ -0,0 +1,259 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Any, Dict, Optional
+
+from loguru import logger
+from omegaconf import OmegaConf
+from pipecat.audio.vad.silero import VADParams
+
+from nemo.agents.voice_agent.pipecat.services.nemo.diar import NeMoDiarInputParams
+from nemo.agents.voice_agent.pipecat.services.nemo.stt import NeMoSTTInputParams
+
+
+class ConfigManager:
+    """
+    Manages configuration for the voice agent server.
+    Handles loading, merging, and providing access to all configuration parameters.
+    """
+
+    def __init__(self, server_base_path: Optional[str] = None, server_config_path: Optional[str] = None):
+        """
+        Initialize the configuration manager.
+
+        Args:
+            config_path: Path to the main server configuration file.
+                        If None, uses default path from environment variable.
+        """
+        self._server_base_path = server_base_path
+        if server_config_path is not None:
+            self._server_config_path = server_config_path
+        else:
+            self._server_config_path = f"{os.path.abspath(self._server_base_path)}/server_configs/default.yaml"
```

Comment:
> server_base_path: Optional[str] = None

If `server_base_path` is None, it will make `self._server_config_path` an invalid path. We should not allow `server_base_path` to be None